src/colorizer/IRenderCanvas.ts

```diff
@@ -122,3 +123,26 @@ export interface IRenderCanvas {
    */
   getIdAtPixel(x: number, y: number): PixelIdInfo | null;
 }
+
+export interface IInnerRenderCanvas extends IRenderCanvas {
+  /**
+   * Sets a callback function that will be called whenever the canvas is
+   * rendered.
+   */
+  setOnRenderCallback(callback: null | (() => void)): void;
+
+  /**
+   * Returns a Matrix4 that projects from a 3D coordinate (in frame
+   * pixels/volume voxels) to a 2D canvas pixel coordinate, where (0,0) is the
+   * top left corner of the canvas.
+   */
+  getScreenSpaceMatrix(): Matrix4;
+
+  /**
+   * Returns a function that converts a depth value for some object to the
+   * opacity (when covered) and scale of an annotation marker that should be
+   * rendered over it.
+   * @param screenSpaceMatrix The screen space matrix that projects from 3D coordinates to
+   */
+  getDepthToScaleFn(screenSpaceMatrix: Matrix4): (depth: number) => { scale: number; clipOpacity: number };
+}
```

Comment:
I didn't think it made sense to implement these on `CanvasOverlay`, which implements the `IRenderCanvas` type.